internal/controllers/clusterextension_controller.go

```diff
@@ -267,21 +267,13 @@ func (r *ClusterExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alp
 	l.V(1).Info("unpacking resolved bundle")
 	unpackResult, err := r.Unpacker.Unpack(ctx, bundleSource)
 	if err != nil {
-		setStatusUnpackFailed(ext, err.Error())
 		// Wrap the error passed to this with the resolution information until we have successfully
 		// installed since we intend for the progressing condition to replace the resolved condition
 		// and will be removing the .status.resolution field from the ClusterExtension status API
 		setStatusProgressing(ext, wrapErrorWithResolutionInfo(resolvedBundleMetadata, err))
 		return ctrl.Result{}, err
 	}
 
-	switch unpackResult.State {
-	case rukpaksource.StateUnpacked:
-		setStatusUnpacked(ext, unpackResult.Message)
-	default:
-		panic(fmt.Sprintf("unexpected unpack state %q", unpackResult.State))
-	}
```

Comment:
I'm not sure we should entirely remove the panic here because the API of `r.Unpacker.Unpack` doesn't have any guarantees a `nil` error returned means that the unpack was successful and will have the bundle data populated in the result returned. I would consider it out of scope for this PR to refactor the `Unpacker` API, but I would still want us to catch any programmer errors possible from the current state of the API contract.

Instead of removing the switch statement, I think changing it to something like this allows us to catch programmer errors:
```go
if unpackResult.State != rukpaksource.StateUnpacked {
	panic(fmt.Sprintf("unexpected unpack state %q", unpackResult.State))
}
```

Reply:
yeah should have figured the default logic stays will update